t/t-push.sh

```diff
@@ -820,3 +820,16 @@ begin_test "push custom reference"
   assert_server_object "$reponame" "$oid"
 )
 end_test
+
+begin_test "push --object-id (invalid value)"
+(
+  set -e
+
+  push_all_setup "everything"
+
+  git lfs push --object-id origin '' 2>&1 | tee push.log
+  git lfs push --object-id origin "${oid1:1:4}" 2>&1 | tee -a push.log
```

Comment:
I was intrigued by how this works, because the `${oid1:1:4}` would seem to me to be generating a 4-character string starting with the 2nd character in `$oid1`, and that didn't jive with the `len(oid) < 4` check added in `ObjectPath()`.

Well, therein lies a tale.  I've put up two extra commits in my `chrisd8088/empty-oid-push` [branch](https://github.com/chrisd8088/git-lfs/tree/empty-oid-push) which explain it in detail, but the gist is that `push_all_setup()` appears to have been accidentally short-circuiting all the tests which call it (except the first one), and when that bug is fixed, this new test starts failing because it generates a 4-character string which doesn't trigger a second `too short object ID` message from `git lfs push`, as I'd idly puzzled over in the first place.

If you're willing to pull those two commits (f9381fa37cb25af17c6d90f4eea054770cb5692d and 63a0b7a388a35fd507468d2474ef63d0c1cc30b2) into this PR, I think that would resolve all my concerns here and get the test fully working.

Reply:
Good catch.  Done.